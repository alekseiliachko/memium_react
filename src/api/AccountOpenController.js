import APIController from "./ApiController";

class АccountOpenController extends APIController {
  constructor() {
    super();
  }

  async getAccountsDetails(accountId) {
    const details = await this.request(
      "get",
      `/open/account/details/${accountId}`
    );
    return details;
  }

  async getAccountsAvatar(accountId) {
    const avatar = await this.request(
      "get",
      `/open/account/avatar/${accountId}`,
      {
        responseType: "blob",
      }
    );
    return URL.createObjectURL(avatar.data);
  }
}

export default new АccountOpenController();
