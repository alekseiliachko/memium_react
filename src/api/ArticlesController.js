import APIController from "./ApiController";

class AccountsController extends APIController {
  constructor() {
    super();
  }

  async createArticle(articleData) {
    return await this.request("post", "article/", articleData).then(
      (val) => val.data
    );
  }

  async updateArticle(newArticleData) {
    return await this.request("put", "article/", newArticleData);
  }

  async articleHasLike(articleId) {
    return await this.request("get", `article/haslike/${articleId}`);
  }

  async setImageForArticle(articleId, image) {
    return await this.request("post", `article/image/${articleId}`, image).then(
      (val) => val.data
    );
  }

  async myArticles() {
    return await this.request("get", "article/my");
  }

  async deleteArticle(articleId) {
    return await this.request("delete", `article/my/${articleId}`);
  }

  async getArticleById(articleId) {
    return await this.request("get", `/open/article/articleId/${articleId}`);
  }

  async getArticleImage(articleId) {
    const image = await this.request(
      "get",
      `/open/article/image/${articleId}`,
      {
        responseType: "blob",
      }
    );
    if (!image.data.size) {
      return null;
    }
    return URL.createObjectURL(image.data);
  }
}

export default new AccountsController();
