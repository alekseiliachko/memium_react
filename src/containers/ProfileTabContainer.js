import { connect } from "react-redux";
import { ProfileTab } from "../components/ProfileTab";
import {
  loadDetails,
  loadSubs,
  loadAvatar,
  updateDetails,
  updateAvatar,
} from "../redux/user/actions";

const mapStateToProps = (store) => ({});

const mapDispatchToProps = (dispatch) => ({});

export const ProfileTabContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(ProfileHeader);
