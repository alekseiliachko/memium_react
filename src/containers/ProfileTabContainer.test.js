import React from "react";
import renderer from "react-test-renderer";
import { withStore } from "../redux/testHelper";
import reactRouter, { MemoryRouter } from "react-router";
import { ProfileTabContainer } from "./ProfileTabContainer";
import { loadBl, loadLikedList, loadSubs } from "../redux/user/actions";
import { testData2 } from "../redux/testData2";
import AccountOpenController from "../api/AccountOpenController";
import ArticlesController from "../api/ArticlesController";

jest.mock("../api/AccountOpenController");
jest.mock("../api/ArticlesController");

jest.mock("../redux/user/actions", () => {
  return {
    loadBl: jest.fn(),
    loadLikedList: jest.fn(),
    loadSubs: jest.fn(),
  };
});

describe("profile tab container", () => {
  it("renders correctly without data tab 2", async () => {
    const Comp = withStore();
    let tree;

    reactRouter.useLocation = jest.fn().mockReturnValue({ search: "?tab=2" });
    loadBl.mockReturnValue({ type: "who_cares" });
    loadLikedList.mockReturnValue({ type: "who_cares" });
    loadSubs.mockReturnValue({ type: "who_cares" });

    await renderer.act(async () => {
      tree = renderer.create(
        <Comp>
          <MemoryRouter>
            <ProfileTabContainer />
          </MemoryRouter>
        </Comp>
      );
      await Promise.resolve();
    });

    expect(tree.toJSON()).toMatchSnapshot();
    expect(loadBl).toBeCalledTimes(1);
    expect(loadLikedList).toBeCalledTimes(1);
    expect(loadSubs).toBeCalledTimes(1);
  });

  it("renders correctly tab one", async () => {
    const Comp = withStore(testData2);
    let tree;

    reactRouter.useLocation = jest.fn().mockReturnValue({ search: "?tab=1" });

    loadBl.mockReturnValue({ type: "who_cares" });
    loadLikedList.mockReturnValue({ type: "who_cares" });
    loadSubs.mockReturnValue({ type: "who_cares" });

    AccountOpenController.getAccountsAvatar.mockResolvedValue("");
    AccountOpenController.getAccountsDetails.mockResolvedValue({
      data: { name: "" },
    });
    ArticlesController.getArticleImage.mockResolvedValue("");

    await renderer.act(async () => {
      tree = renderer.create(
        <Comp>
          <MemoryRouter>
            <ProfileTabContainer />
          </MemoryRouter>
        </Comp>
      );
      await Promise.resolve();
    });

    expect(tree.toJSON()).toMatchSnapshot();
    expect(loadBl).toBeCalledTimes(1);
    expect(loadLikedList).toBeCalledTimes(1);
    expect(loadSubs).toBeCalledTimes(1);
  });

  it("renders correctly tab zero", async () => {
    const Comp = withStore(testData2);
    let tree;

    reactRouter.useLocation = jest.fn().mockReturnValue({ search: "?tab=0" });

    loadBl.mockReturnValue({ type: "who_cares" });
    loadLikedList.mockReturnValue({ type: "who_cares" });
    loadSubs.mockReturnValue({ type: "who_cares" });

    await renderer.act(async () => {
      tree = renderer.create(
        <Comp>
          <MemoryRouter>
            <ProfileTabContainer />
          </MemoryRouter>
        </Comp>
      );
      await Promise.resolve();
    });

    expect(tree.toJSON()).toMatchSnapshot();
    expect(loadBl).toBeCalledTimes(1);
    expect(loadLikedList).toBeCalledTimes(1);
    expect(loadSubs).toBeCalledTimes(1);
  });

  it("renders correctly tab two", async () => {
    const Comp = withStore(testData2);
    let tree;

    reactRouter.useLocation = jest.fn().mockReturnValue({ search: "?tab=2" });

    loadBl.mockReturnValue({ type: "who_cares" });
    loadLikedList.mockReturnValue({ type: "who_cares" });
    loadSubs.mockReturnValue({ type: "who_cares" });

    await renderer.act(async () => {
      tree = renderer.create(
        <Comp>
          <MemoryRouter>
            <ProfileTabContainer />
          </MemoryRouter>
        </Comp>
      );
      await Promise.resolve();
    });

    expect(tree.toJSON()).toMatchSnapshot();
    expect(loadBl).toBeCalledTimes(1);
    expect(loadLikedList).toBeCalledTimes(1);
    expect(loadSubs).toBeCalledTimes(1);
  });
});
