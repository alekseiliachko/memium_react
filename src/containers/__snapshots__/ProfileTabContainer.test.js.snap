// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile tab container renders correctly tab one 1`] = `
<div>
  <div
    className="MuiTabs-root WithStyles(ForwardRef(Tabs))-root-1"
  >
    <div
      className="MuiTabs-scroller MuiTabs-fixed"
      onScroll={[Function]}
      style={
        Object {
          "marginBottom": null,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-label="ant example"
        className="MuiTabs-flexContainer"
        onKeyDown={[Function]}
        role="tablist"
      >
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root Component-root-3 MuiTab-textColorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Черный список
          </span>
        </button>
        <button
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root Component-root-3 MuiTab-textColorInherit Mui-selected"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Понравившиеся
          </span>
        </button>
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root Component-root-3 MuiTab-textColorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Подписки
          </span>
        </button>
      </div>
      <span
        className="PrivateTabIndicator-root-4 PrivateTabIndicator-colorSecondary-6 MuiTabs-indicator WithStyles(ForwardRef(Tabs))-indicator-2"
        style={
          Object {
            "left": 0,
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-23 MuiPaper-outlined MuiPaper-rounded"
    onClick={[Function]}
  >
    <div
      className="MuiBox-root MuiBox-root-27"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-24 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <div
        className="MuiBox-root MuiBox-root-28"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
        >
          
        </h6>
        <h6
          className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
        >
          Sat Mar 06 2021
        </h6>
      </div>
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    
    <h5
      className="MuiTypography-root MuiTypography-h5"
    >
      Русские мемчанские
    </h5>
  </div>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-23 MuiPaper-outlined MuiPaper-rounded"
    onClick={[Function]}
  >
    <div
      className="MuiBox-root MuiBox-root-29"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-24 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <div
        className="MuiBox-root MuiBox-root-30"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
        >
          
        </h6>
        <h6
          className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
        >
          Sat Mar 06 2021
        </h6>
      </div>
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    
    <h5
      className="MuiTypography-root MuiTypography-h5"
    >
      Chem?
    </h5>
  </div>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-23 MuiPaper-outlined MuiPaper-rounded"
    onClick={[Function]}
  >
    <div
      className="MuiBox-root MuiBox-root-31"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-24 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <div
        className="MuiBox-root MuiBox-root-32"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
        >
          
        </h6>
        <h6
          className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
        >
          Sat Mar 06 2021
        </h6>
      </div>
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    
    <h5
      className="MuiTypography-root MuiTypography-h5"
    >
      It?
    </h5>
  </div>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-23 MuiPaper-outlined MuiPaper-rounded"
    onClick={[Function]}
  >
    <div
      className="MuiBox-root MuiBox-root-33"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-24 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <div
        className="MuiBox-root MuiBox-root-34"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
        >
          
        </h6>
        <h6
          className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
        >
          Sat Mar 06 2021
        </h6>
      </div>
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    
    <h5
      className="MuiTypography-root MuiTypography-h5"
    >
      Bio?
    </h5>
  </div>
</div>
`;

exports[`profile tab container renders correctly tab two 1`] = `
<div>
  <div
    className="MuiTabs-root WithStyles(ForwardRef(Tabs))-root-1"
  >
    <div
      className="MuiTabs-scroller MuiTabs-fixed"
      onScroll={[Function]}
      style={
        Object {
          "marginBottom": null,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-label="ant example"
        className="MuiTabs-flexContainer"
        onKeyDown={[Function]}
        role="tablist"
      >
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root Component-root-3 MuiTab-textColorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Черный список
          </span>
        </button>
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root Component-root-3 MuiTab-textColorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Понравившиеся
          </span>
        </button>
        <button
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root Component-root-3 MuiTab-textColorInherit Mui-selected"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Подписки
          </span>
        </button>
      </div>
      <span
        className="PrivateTabIndicator-root-4 PrivateTabIndicator-colorSecondary-6 MuiTabs-indicator WithStyles(ForwardRef(Tabs))-indicator-2"
        style={
          Object {
            "left": 0,
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-8 MuiPaper-outlined MuiPaper-rounded"
  >
    <div
      className="MuiBox-root MuiBox-root-41"
    >
      <div
        className="MuiBox-root MuiBox-root-42"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-9 MuiAvatar-colorDefault"
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiAvatar-fallback"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-43"
      >
        <div
          className="MuiCardContent-root makeStyles-content-11"
        >
          <h2
            className="MuiTypography-root makeStyles-name-10 MuiTypography-h5"
          />
          <p
            className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            populated
          </p>
        </div>
        <div
          className="MuiCardActions-root makeStyles-actions-12 MuiCardActions-spacing"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-13"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Unsub
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`profile tab container renders correctly tab zero 1`] = `
<div>
  <div
    className="MuiTabs-root WithStyles(ForwardRef(Tabs))-root-1"
  >
    <div
      className="MuiTabs-scroller MuiTabs-fixed"
      onScroll={[Function]}
      style={
        Object {
          "marginBottom": null,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-label="ant example"
        className="MuiTabs-flexContainer"
        onKeyDown={[Function]}
        role="tablist"
      >
        <button
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root Component-root-3 MuiTab-textColorInherit Mui-selected"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Черный список
          </span>
        </button>
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root Component-root-3 MuiTab-textColorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Понравившиеся
          </span>
        </button>
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root Component-root-3 MuiTab-textColorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Подписки
          </span>
        </button>
      </div>
      <span
        className="PrivateTabIndicator-root-4 PrivateTabIndicator-colorSecondary-6 MuiTabs-indicator WithStyles(ForwardRef(Tabs))-indicator-2"
        style={
          Object {
            "left": 0,
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-8 MuiPaper-outlined MuiPaper-rounded"
  >
    <div
      className="MuiBox-root MuiBox-root-35"
    >
      <div
        className="MuiBox-root MuiBox-root-36"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-9 MuiAvatar-colorDefault"
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiAvatar-fallback"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-37"
      >
        <div
          className="MuiCardContent-root makeStyles-content-11"
        >
          <h2
            className="MuiTypography-root makeStyles-name-10 MuiTypography-h5"
          />
          <p
            className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            populated
          </p>
        </div>
        <div
          className="MuiCardActions-root makeStyles-actions-12 MuiCardActions-spacing"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-13"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Unblock
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`profile tab container renders correctly without data tab 2 1`] = `
<div>
  <div
    className="MuiTabs-root WithStyles(ForwardRef(Tabs))-root-1"
  >
    <div
      className="MuiTabs-scroller MuiTabs-fixed"
      onScroll={[Function]}
      style={
        Object {
          "marginBottom": null,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-label="ant example"
        className="MuiTabs-flexContainer"
        onKeyDown={[Function]}
        role="tablist"
      >
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root Component-root-3 MuiTab-textColorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Черный список
          </span>
        </button>
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root Component-root-3 MuiTab-textColorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Понравившиеся
          </span>
        </button>
        <button
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root Component-root-3 MuiTab-textColorInherit Mui-selected"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Подписки
          </span>
        </button>
      </div>
      <span
        className="PrivateTabIndicator-root-4 PrivateTabIndicator-colorSecondary-6 MuiTabs-indicator WithStyles(ForwardRef(Tabs))-indicator-2"
        style={
          Object {
            "left": 0,
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-8 MuiPaper-outlined MuiPaper-rounded"
  >
    <div
      className="MuiBox-root MuiBox-root-17"
    >
      <div
        className="MuiBox-root MuiBox-root-18"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-9 MuiAvatar-colorDefault"
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiAvatar-fallback"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-19"
      >
        <div
          className="MuiCardContent-root makeStyles-content-11"
        >
          <h2
            className="MuiTypography-root makeStyles-name-10 MuiTypography-h5"
          />
          <p
            className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            populated
          </p>
        </div>
        <div
          className="MuiCardActions-root makeStyles-actions-12 MuiCardActions-spacing"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-13"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Unsub
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
